department-of-veterans-affairs/caseflow
# Pagination Prep Hearings APP

### Description
This is a refactor of the `HearingDayRange` mainly of `#upcoming_days_for_judge` making the Hearings index page for judges substantially more efficient. I have also refactored the main method used for all non-vso, non-judge users.

### Acceptance Criteria
- [ ] The Hearings index page /hearings/schedule should load for all users (VSO, judges and all the rest)

### Testing Plan
1. Shadow user BVARERDMAN
2. Go to /hearings/schedule The page should load the HearingDays
3. Shadow user BILLIE_VSO
4. Go to /hearings/schedule The page should load the HearingDays
5. Shadow user BVASYELLOW
6. Go to /hearings/schedule The page should load the HearingDays

In addition to testing it locally I wrote a script which I ran in production to verified that judges are getting the same hearings with the new query. https://hackmd.io/8Q3IjHWbR4WR8OUyCBdYlA
